win_domain - Add ability to specify netbios name

jtauke opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

<!--- If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.-->

Changes are to win_domain.ps1 and win_domain.py.  The changes allow a user to specify the netbios name of the newly created domain.  This is helpful when following best practice from microsoft and specifying an internal domain for Active Directory (i.e. ad.test.com or int.test.com).  Prior to this change the netbios name of the domain would default to AD or INT using the examples above.  After this change, the netbios name would be TEST.

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module, plugin, module or task -->
win_domain.ps1
win_domain.py
##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.5.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/joe.tauke/Dev/ansible/lib/ansible/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.14 (default, Mar 14 2018, 13:36:31) [GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]

```


##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful. -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

Commit:
Added netbios option to win_domain.ps1 and updated documentation

jtauke committed:
formatting change

jtauke committed:
formatting change

jtauke committed:
adding version added line for domain_netbios_name

jtauke commented:
ready_for_review

jhawkesworth commented on lib/ansible/modules/windows/win_domain.ps1:
```diff
@@ -80,7 +81,10 @@ If(-not $forest) {
         }
-
+	if ($domain_netbios_name) {
+        $install_forest_args.DomainNetBiosName = $domain_netbios_name
```
I know this is super-fussy, but I would prefer it if this line was indented to match the other if blocks in the module.

jtauke committed:
Identation fix

jtauke commented:
Indentation should be properly set now.

jhawkesworth commented:
Thanks for fixing the indentation.  Sorry I just thought of this, and I know its only a litle thing... but it would be good to describe what happens if the netbios name is not specified in the documentation.  Something like...  Default: same as the first part of the dns_domain_name, up to, but not including the '.'  (feel free to describe it better than this).

Otherwise I think this is ready to be merged.

jborean93 commented on lib/ansible/modules/windows/win_domain.py:
```diff
@@ -21,6 +21,11 @@
+  domain_netbios_name:
+    description:
+      - the netbios name of the domain
+    required: no
```
`required: no` is the default and we usually just omit it.

jborean93 commented on lib/ansible/modules/windows/win_domain.py:
```diff
@@ -21,6 +21,11 @@
     required: yes
+  domain_netbios_name:
+    description:
+      - the netbios name of the domain
```
I know the other parts of the documentation isn't in a sentence (we should fix this) but usually the description is capitalized and ends with `.`

jborean93 reviewed (commented):
Just minor nitpicks, I agree that the description for the new option can be expanded a bit.

jtauke committed:
Clarity fixes for descriptions

Changes to description for netbios_domain_name and uniformity changes to other documentation sections (bringing them in line with other sections of the document).

jtauke commented:
@jborean93 and @jhawkesworth, thanks for the review comments. I just committed some alterations around the description for domain_netbios_name (as well as uniformity changes to bring other sections in line with sentence format).

jborean93 committed:
Fix minor indentation

jborean93 reviewed (approved):
Thanks @jtauke
